src/core/markovProcessGraph.js

```diff
@@ -644,33 +644,64 @@ export class MarkovProcessGraph {
     return this._participants;
   }
 
+  /**
+   * Returns a canonical ordering of the nodes in the graph.
+   *
+   * No assumptions should be made about the node order, other than
+   * that it is stable for any given MarkovProcessGraph.
+   */
+  nodeOrder(): $ReadOnlyArray<NodeAddressT> {
+    return Array.from(this._nodes.keys()).sort();
+  }
+
   node(address: NodeAddressT): MarkovNode | null {
     NodeAddress.assertValid(address);
     return this._nodes.get(address) || null;
   }
 
+  /**
+   * Iterate over the nodes in the graph. If a prefix is provided,
+   * only nodes matching that prefix will be returned.
+   *
+   * The nodes are always iterated over in the node order.
+   */
   *nodes(options?: {|+prefix: NodeAddressT|}): Iterator<MarkovNode> {
     const prefix = options ? options.prefix : NodeAddress.empty;
-    for (const node of this._nodes.values()) {
-      if (NodeAddress.hasPrefix(node.address, prefix)) {
-        yield node;
+    for (const address of this.nodeOrder()) {
+      if (NodeAddress.hasPrefix(address, prefix)) {
+        yield NullUtil.get(this.node(address));
```

Comment:
Follow-up to above: just iterate over `this._nodes` here and pull out
`[address, value]` rather than forcing another `this.node` lookup and
`NullUtil.get` (the lookup is more important, since it hits memory).